### PR TITLE
GF-37 - Server side rendering for Constructors and Drivers

### PR DESCRIPTION
**Pull request by MIKALAKE**

### CODE REVIEW

File: src/show/pages/public/Constructor/index.jsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+const Constructor = () => {
+  const { id } = useParams();
+  const [getConstructorData, setGetConstructorData] = useState({});
```
Comment (iamalextodoran) [inline src/show/pages/public/Constructor/index.jsx]: Not a good name for those.

File: src/show/pages/public/Constructor/index.jsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  useEffect(() => {
+    axios
+      .get(`http://localhost:3000/v1/constructors/${id}`)
+      .then(res => setGetConstructorData(res.data))
+      .catch(err => err);
+  }, [id]);
+
+  return (
+    <div className='bg-white-600 w-full h-full'>
+      {getConstructorData && (
```
Comment (iamalextodoran) [inline src/show/pages/public/Constructor/index.jsx]: Nice! 👍🏼

File: src/show/pages/public/Driver/index.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import Moment from 'react-moment';
```
Comment (iamalextodoran) [inline src/show/pages/public/Driver/index.jsx]: Group with the ones above.

File: src/show/pages/public/Driver/index.jsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+const Driver = () => {
+  const { id } = useParams();
+  const [getDriverData, setGetDriverData] = useState({});
```
Comment (iamalextodoran) [inline src/show/pages/public/Driver/index.jsx]: Not a good name

File: src/show/pages/public/Home/index.jsx [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
   ConstructorCard,
   DriverCard,
   UpcomingEvent
 } from '../../../components';
+import { CONSTRUCTOR_PATH } from '../../../../process/routes/paths';
 
 const Home = () => {
   const [constructors, setConstructors] = useState([]);
   const [event, setEvent] = useState({});
+  const navigate = useNavigate();
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.jsx:16]: Empty line before this.

File: src/show/pages/public/Home/index.jsx [hunk 6]
```diff
@@ -18,13 +21,13 @@ const Home = () => {
       .catch(err => err);
 
     axios
-      .get('http://localhost:3000/v1/constructors/')
+      .get(`http://localhost:3000/v1/constructors/`)
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.jsx]: No need for back ticks

File: src/show/pages/public/Home/index.jsx [hunk 7]
```diff
@@ -39,7 +42,16 @@ const Home = () => {
...
           {constructors.length > 0 ? (
             constructors?.map(constructor => (
-              <ConstructorCard key={constructor.id} constructor={constructor} />
+              <div
+                onClick={() =>
+                  navigate(CONSTRUCTOR_PATH.replace(':id', constructor.id))
+                }
+                key={constructor.id}>
+                <ConstructorCard
+                  key={constructor.id}
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.jsx]: There's no more need for this key, only the first item needs it.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

**Comment by iamalextodoran** [comment]
No page for events?

File: src/show/pages/public/Home/index.jsx [hunk 8]
```diff
@@ -18,13 +21,13 @@ const Home = () => {
       .catch(err => err);
```
Comment (iamdavidpetric) [inline src/show/pages/public/Home/index.jsx:21]: ???

File: src/show/pages/public/Constructor/index.jsx (hunk 1 above)
Comment (MIKALAKE) [reply src/show/pages/public/Constructor/index.jsx]: It ok to call them just constructorData and setConstructorData?

File: src/show/pages/public/Constructor/index.jsx (hunk 2 above)
Comment (MIKALAKE) [reply src/show/pages/public/Constructor/index.jsx]: thx

File: src/show/pages/public/Driver/index.jsx (hunk 3 above)
Comment (MIKALAKE) [reply src/show/pages/public/Driver/index.jsx]: done

File: src/show/pages/public/Home/index.jsx (hunk 5 above)
Comment (MIKALAKE) [reply src/show/pages/public/Home/index.jsx:16]: done

File: src/show/pages/public/Home/index.jsx (hunk 6 above)
Comment (MIKALAKE) [reply src/show/pages/public/Home/index.jsx]: done

File: src/show/pages/public/Home/index.jsx (hunk 7 above)
Comment (MIKALAKE) [reply src/show/pages/public/Home/index.jsx]: done

**Comment by MIKALAKE** [comment]
> No page for events?

No, because the Upcoming Event is just for informational purpose, i will make a page for event details when the Schedule page is done

**Review by iamalextodoran — APPROVED** [review]

**Pull request merged**